### ISSUE TITLE
Consider adding support for direct transformation of navis.TreeNeuron objects

### ISSUE THREAD

**Issue by SridharJagannathan** [schlegelp/navis#14]
For example, when I want to transform an object from say FANC (female nerve cord) space to MANC (male nerve cord) space,
I would first to the following:
1) Build a thin-plate spline registration
`fanc2manctr=navis.transforms.thinplate.TPStransform(fancreg_pts, mancreg_pts)`
2. Transform a neuron
    ```
    fanc2mancskeleton = fancskeleton.copy()
   fanc2mancskeleton.nodes[['x','y','z']] = fanc2manctr.xform(fancskeleton.nodes[['x','y','z']])
   ```
Here you can see that I will be able to transform neuron `fancskeleton` to `fanc2mancskeleton` using the registration.
But it will be easier for users to directly transform say,
 `fanc2mancskeleton = fanc2manctr.xform(fancskeleton)`
Thus, if you have an implementation of transform class that will also work on neuron and neuronlist objects (this is the way it is currently implemented in `nat`)

**Comment by schlegelp** [comment]
The transforms classes are the lowest level entry point and are meant to be as lean and mean as possible - so they will remain as they are.

Transforms in `nat` don't do much by themselves either: the lowest entry point is `xform_points` which takes points and a transform (analogous to `TPStransform.xform`). One up is `xform` that takes various data and transforms, turns the data into points and calls `xform_points`. And finally there is `xform_brain` which finds the correct transform (sequence) for you and then calls `xform` into `xform_points`. 

I will think about adding `navis.xform` to match `nat::xform`.

**Comment by schlegelp** [comment]
`navis.xform` implemented with 08a8ae6d3bb5fdec0e334b75be1e700ded94f79c. From your example:

```Python
fanc2manctr=navis.transforms.thinplate.TPStransform(fancreg_pts, mancreg_pts)
fanc2mancskeleton = navis.xform(fancskeleton, transform=fanc2manctr)
```

I will aim to push a new version to PyPI this weekend. In the meantime, you would need to install/update from Github to test the changes.

**Comment by schlegelp** [comment]
Closing this issue as seems solved.
